assets/src/components/activities/adaptive/AdaptiveDelivery.tsx

```diff
@@ -89,7 +89,13 @@ const Adaptive = (props: DeliveryElementProps<AdaptiveModelSchema>) => {
       }, 2000);
     });
     sharedPromiseMap.set(props.model.id, { promise, resolve, reject });
-
+    if (parts.length == 0 && resolve) {
```

Comment:
I think we can move this to the top of this method and don't even need to setup the promise stuff if there are no parts
```
if (!parts.length) {
  if (props.onReady) {
    props.onReady(attemptState.attemptGuid);
  }
  return;
 }
```

Reply:
Done